server/Websocket/websocketserver.mjs

```diff
@@ -1,109 +1,202 @@
+  const clientKey = req.headers["sec-websocket-key"];
+  if (!clientKey) {
+    socket.write("HTTP/1.1 400 Bad Request\r\n\r\n");
+    socket.destroy();
+    return;
+  }
+
+  const acceptKey = createAcceptKey(clientKey);
+  const responseHeaders = [
+    "HTTP/1.1 101 Switching Protocols",
+    "Upgrade: websocket",
+    "Connection: Upgrade",
+    `Sec-WebSocket-Accept: ${acceptKey}`,
+    "\r\n",
+  ].join("\r\n");
+
+  socket.write(responseHeaders);
+
+  // buffer incoming data here
+  socket._buffer = Buffer.alloc(0);
+
+  socket.on("data", (data) => {
+    socket._buffer = Buffer.concat([socket._buffer, data]);
+    processBuffer(socket);
   });
 
-  // Add event listener for disconnections
-  socket.on('end', () => {
-    console.log('Client disconnected');
+  socket.on("error", (err) => {
+    console.error("Socket error:", err);
   });
-}
 
-// combining the client's key with the magic string, hashing it, and returning a base64-encoded result.
-function createSocketAccept(webClientSocketKey){
-  const hash = crypto.createHash('sha1').update(webClientSocketKey + WEBSOCKET_MAGIC_STRING);
-  const response = hash.digest('base64');
-  return response;
+  socket.on("end", () => {
+    console.log("Client disconnected");
+  });
 }
 
-function prepareHandShakeResponse(webClientSocketKey) {
-  const accept = createSocketAccept(webClientSocketKey);
-  const headers = {
-    'HTTP/1.1 101 Switching Protocols': null,
-    'Upgrade': 'websocket',
-    'Connection': 'Upgrade',
-    'Sec-WebSocket-Accept': accept
-  };
-  // convert headers to string because socket.write() expects a string
-  const headerString = Object.keys(headers).map(key => `${key}: ${headers[key]}`).join('\r\n') + '\r\n\r\n';
-  return headerString;
+function createAcceptKey(key) {
+  return crypto
+    .createHash("sha1")
+    .update(key + WEBSOCKET_MAGIC_STRING)
+    .digest("base64");
 }
 
+// dataframe processing
+function processBuffer(socket) {
+  let buffer = socket._buffer;
 
-function readSocketData(socket) {
-  // consume optcode (first byte)
-  socket.read(1);
+  while (true) {
+    if (buffer.length < 2) return; // need at least 2 bytes to read header
 
-  const [payloadLength] = socket.read(1);
+    const firstByte = buffer[0];
+    const secondByte = buffer[1];
 
-  // first bit is always 1, so we can ignore it (1 bit = 128)
-  const LengthIndicator = payloadLength - FIRST_BIT;
+    const fin = (firstByte & 0x80) !== 0;
+    const opcode = firstByte & 0x0f;
+    const masked = (secondByte & 0x80) !== 0;
+    let payloadLength = secondByte & 0x7f;
 
-  let messageLength = 0;
+    let offset = 2;
 
-  if(LengthIndicator <= MAX_7_BITS_INT_MARKER){
-    messageLength = LengthIndicator;
-  } else {
-    console.error("Message length not supported right now");
-  }
+    if (payloadLength === 126) {
+      if (buffer.length < offset + 2) return;
+      payloadLength = buffer.readUInt16BE(offset);
+      offset += 2;
+    } else if (payloadLength === 127) {
```

Comment:
No "magic" numbers, please define constants such as `126`, `127`, etc... in a constant above or constants file.